SaraAlert/SaraAlert
# SARAALERT-1667: Make vaccine value sets configurable

# Description
Jira Ticket: [SARAALERT-1667](https://tracker.codev.mitre.org/browse/SARAALERT-1667)

To allow for the monitoring of new diseases in the future, the configuration in `vaccines.yml` should be used to populate the Advanced Filter options, rather than the static data in `app/javascript/data/advancedFilterOptions.js`.

As per Slack conversation [here](https://mitrecorp.slack.com/archives/C029CLK0X2P/p1645475659046289), adding downselecting behavior within Vaccine (Combination) Advanced Filter is out of scope for this ticket. 

## Demo/Screenshots

**Advanced filter shows "Another Condition" in addition to COVID-19 within vaccine group.** 
<img width="1111" alt="Screen Shot 2022-03-01 at 6 59 29 AM" src="https://user-images.githubusercontent.com/2328582/156165694-30a8db62-0c62-4469-9f56-3bc279e13c6e.png">

**Advanced filter shows additional product names within product name select.** 
<img width="1249" alt="Screen Shot 2022-03-01 at 6 59 38 AM" src="https://user-images.githubusercontent.com/2328582/156165844-1e0025a6-2297-4621-a5b1-d04643455a98.png">

**A saved filter named, "another condition filter," that pertains to "Another Condition"**  
<img width="1112" alt="Screen Shot 2022-03-01 at 6 59 51 AM" src="https://user-images.githubusercontent.com/2328582/156165920-1a19fd8f-2c3b-4588-b2ec-c52270b876dc.png">

**Custom Export using an Advanced Filter that pertains to "Another Condition"**  
<img width="1221" alt="Screen Shot 2022-03-01 at 7 00 11 AM" src="https://user-images.githubusercontent.com/2328582/156166022-d2c87f43-f296-4dbb-917f-38041f007d14.png">

**Patient page and "Add New Vaccination" modal that pertains to "Another Condition"**
<img width="917" alt="Screen Shot 2022-03-01 at 7 00 36 AM" src="https://user-images.githubusercontent.com/2328582/156166195-8cad5b05-4e06-49d0-a478-7448a14db255.png">

## Important Changes
Please list important files (meaning substantial or integral to the PR) along with a list of the general changes that should be highlighted for reviewers.

`app/javascript/components/public_health/query/AdvancedFilter.js`
- Adds result from `getVaccineAdvancedFilters` utility method to `advancedFilterOptions` (if not yet added)

`app/javascript/data/advancedFilterOptions.js`
- Removes 'vaccination' data

`app/javascript/tests/mocks/mockVaccineStandards.js`
- Adds mock for vaccine standards data

`app/javascript/tests/public_health/query/AdvancedFilter.test.js`
- Uses mock for vaccine standards data

`app/javascript/utils/VaccineAdvancedFilters.js`
- Transforms the vaccine_standards configuration object in `/config/vaccines.yml` to match the structure in `app/javascript/data/advancedFilterOptions.js`

`config/initializers/vaccine_standards.rb`
- Adds `VACCINE_STANDARDS` global

`config/vaccines.yml`
- Adds comment around safe and unsafe changes to vaccine configuration


## Testing Recommendations

- Add another condition to `/config/vaccines.yml`
- Confirm that all data related to the newly-added condition appears in AdvancedFilter Vaccine (Combination) (i.e., vaccine group, product name, dose number)
- Confirm that saving an AdvancedFilter with the newly-added condition works as expected
- Confirm that all data related to the newly-added condition appears in Advanced Filter Vaccine (Combination) within Custom Export
- Confirm that saving a Custom Export using this AdvancedFilter works as expected
- Confirm that adding a new vaccination on the Patient page with the newly-added condition works as expected
- Confirm that adding a new vaccination on the Patient page with the newly-added condition exhibits downselecting (i.e., when I select the newly-added condition for the Vaccine Group, I only see data pertaining to the newly-added condition within Product Name and Dose Number)
- Confirm that the comments added to `config/vaccines.yml` around safe and unsafe changes to the vaccine configuration are clear 

# Checklists

**Submitter:**


@sgober  :


@tstrass  :


@timwongj  :
